process/timeout.rs

```diff
@@ -0,0 +1,435 @@
+//
+// Copyright (c) 2024 Hemi Labs, Inc.
+//
+// This file is part of the posixutils-rs project covered under
+// the MIT License.  For the full license text, please see the LICENSE
+// file in the root directory of this project.
+// SPDX-License-Identifier: MIT
+//
+
+use clap::Parser;
+use gettextrs::{bind_textdomain_codeset, setlocale, textdomain, LocaleCategory};
+use nix::{
+    errno::Errno,
+    sys::{
+        resource::{setrlimit, Resource},
+        signal::{
+            raise, sigaction, signal, sigprocmask, SaFlags, SigAction, SigHandler, SigSet,
+            SigmaskHow,
+            Signal::{self, SIGKILL, SIGTERM, SIGTTIN, SIGTTOU},
+        },
+        wait::{waitpid, WaitPidFlag, WaitStatus},
+    },
+    unistd::{alarm, execvp, fork, setpgid, ForkResult, Pid},
+};
+use plib::PROJECT_NAME;
+use std::{
+    error::Error,
+    ffi::CString,
+    str::FromStr,
+    sync::{
+        atomic::{AtomicBool, AtomicI32, Ordering},
+        Arc, LazyLock, Mutex,
+    },
+    time::Duration,
+};
+
+static FOREGROUND: LazyLock<Arc<AtomicBool>> = LazyLock::new(|| Arc::new(AtomicBool::new(false)));
+static FIRST_SIGNAL: LazyLock<Arc<AtomicI32>> =
+    LazyLock::new(|| Arc::new(AtomicI32::new(SIGTERM as i32)));
+static KILL_AFTER: LazyLock<Arc<Mutex<Option<Duration>>>> =
+    LazyLock::new(|| Arc::new(Mutex::new(None)));
+static MONITORED_PID: LazyLock<Arc<AtomicI32>> = LazyLock::new(|| Arc::new(AtomicI32::new(0)));
+static TIMED_OUT: LazyLock<Arc<AtomicBool>> = LazyLock::new(|| Arc::new(AtomicBool::new(false)));
```

Comment:
`Arc`s here are redundant. You're not cloning anywhere since everything is already global. You don't need to wrap an atomic value in an atomic ref counter for it to be thread safe. Atomic values are enough

Reply:
I didn't understood fully if I need to remove `Arc<Mutex<Option<Duration>>>` for `KILL_AFTER`.

Reply:
I've tested it locally and looks like it's fine

Reply:
As I understood, even `LazyLock` is redundant, but `Mutex` for `Option<Duration>` is necessary, because `KILL_AFTER` need to be mutable.

Reply:
Nice!